app/src/main/java/org/wikipedia/games/onthisday/OnThisDayGameActivity.kt

```diff
@@ -398,14 +435,29 @@ class OnThisDayGameActivity : BaseActivity() {
         }, 500)
     }
 
-    fun openArticleBottomSheet(pageSummary: PageSummary, callback: BottomSheetBehavior.BottomSheetCallback) {
+    fun openArticleBottomSheet(pageSummary: PageSummary, updateBookmark: () -> Unit) {
         if (viewModel.gameState.value !is OnThisDayGameViewModel.GameEnded) {
             return
         }
+        binding.root.requestApplyInsets()
```

Comment:
Calling `requestApplyInsets()` does not guarantee that insets will be applied. On some API versions, this will _not_ call the insetsListener, and your insets will remain `null`, causing a crash.

This is starting to feel really complex...

Are we sure we can't convince Design to make this component into a proper BottomSheetDialogFragment, and make it behave like a BottomSheetDialog is supposed to behave?
Something like this, where the content flows vertically, and naturally ends when it ends:

![image](https://github.com/user-attachments/assets/afdeacd3-980a-4b2c-94a5-a2628836e95a)